library/firewall_lib_facts.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2016,2017,2020,2021 Red Hat, Inc.
+# Reusing some firewalld code
+# Authors:
+# Thomas Woerner <twoerner@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: firewall_lib_facts
+short_description: module for firewall role
+requirements:
+  - python3-firewall or python-firewall
+description:
+  Get firewalld configuration information, including default settings
+  and custom settings. There are no options for this module
+author: Brennan Paciorek (@BrennanPaciorek)
+"""
+
+RETURN = """
+changed:
+  description:
+    whether anything was changed, most likely false
+  type: bool
+  returned: always
+  sample: False
+ansible_facts:
+  description:
+    any facts that are returned via this module running
+    currently only gets firewalld configuration
+    when calling the module
+  type: dict
+  returned: always
+  sample: {
+      "config": {
+          "default": {},
+          "custom": {},
+      },
+  }
+"""
+
+EXAMPLES = """
+# Run with no parameters to gather facts
+firewall_lib_facts:
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+import os
+
+
+def config_to_dict(module):
+    config = {}
+    defaults = {}
+    custom = {}
+
+    defaults_path = os.path.join("/", "usr", "lib", "firewalld")
```

Comment:
https://stackoverflow.com/questions/16010992/how-to-use-directory-separator-in-both-linux-and-windows-in-python/16011057#16011057

Reply:
Is there any purpose in doing so? This role should run exclusively on Linux, so the explicit "/" does not seem like it would be an issue

Although, looking at this, I realized that os.path.join() is probably unfit for what I was trying to achieve here, as I could probably just write out the string "/usr/lib/firewalld"